### PR TITLE
Destroy enemy

### PR DESCRIPTION
**Pull request by Lare1426**
This feature branch merges missiles into when the current displayed hexadecimal matches any of the enemies on the screen. Once that has happened it will save only 1 missile and shoot it up until it hits it's target (enemy).

### CODE REVIEW

File: functions.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from random import randint
+from math import ceil
 import pygame
-from classes import Point, Enemy
+from classes import Point, Triangle, Enemy
```
Comment (studleton) [inline functions.py:4]: `ceil` and `Triangle` are unused.

**Review by studleton — CHANGES_REQUESTED** [review]
Overall quite polished, which mases sense considering the number of iterations and refinement. A good final product of that, which means it was worth it, not even considering what you've learned along the way.

A few minor comments.

File: classes.py [hunk 2]
```diff
@@ -84,7 +144,7 @@ def __init__(self, position, size, font_size, hexadecimals, game):
     self.draw_display()
 
   def update_display(self, binary_boxes):
-    binary = "".join(binary_box.current_bit for binary_box in binary_boxes)
+    binary = reduce(lambda string, box: string + str(int(box.current_bit)), binary_boxes, "")
```
[the remark this answers is not in the document]
Comment (Lare1426) [reply classes.py:147]: Yes

File: classes.py [hunk 3]
```diff
@@ -15,25 +18,79 @@ def __init__(self):
...
+        continue
+
+      step = ceil((0.01 * abs(distance) + 1)**2)
+      if distance < 0: step = -step
+      missile.move(x=step)
+
+      new_missiles.append(missile)
+    
+    if not len(new_missiles):
+      self.missiles = [self.missiles[0]]
```
[the remark this answers is not in the document]
Comment (Lare1426) [reply classes.py:80]: Because then it doesn't have anything to pull from the missiles

**Review by studleton — APPROVED** [review]

**Pull request merged**